src/components/button/template.test.js

```diff
@@ -0,0 +1,254 @@
+      const { $ } = render('button', examples['explicit-button'])
+
+      const $component = $('.govuk-c-button')
+      expect($component.get(0).tagName).toEqual('button')
+      expect($component.attr('type')).toEqual('submit')
+    })
+
+    it('renders with attributes', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        attributes: {
+          'aria-controls': 'example-id',
+          'data-tracking-dimension': '123'
+        },
+        text: 'Save and continue'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('aria-controls')).toEqual('example-id')
+      expect($component.attr('data-tracking-dimension')).toEqual('123')
+    })
+
+    it('renders with classes', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        classes: 'app-c-button--custom-modifier',
+        text: 'Start now'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.hasClass('app-c-button--custom-modifier')).toBeTruthy()
+    })
+
+    it('renders with disabled', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        text: 'Start now',
+        disabled: true
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('aria-disabled')).toEqual('true')
+      expect($component.attr('disabled')).toEqual('disabled')
+      expect($component.hasClass('govuk-c-button--disabled')).toBeTruthy()
+    })
+
+    it('renders with name', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        text: 'Start now',
+        name: 'start-now'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('name')).toEqual('start-now')
+    })
+
+    it('renders with value', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        text: 'Start now',
+        value: 'start'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('value')).toEqual('start')
+    })
+
+    it('renders with html', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        html: 'Start <em>now</em>'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.html()).toContain('Start <em>now</em>')
+    })
+
+    it('renders with type', () => {
+      const { $ } = render('button', {
+        element: 'button',
+        type: 'button',
+        text: 'Start now'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('type')).toEqual('button')
+    })
+  })
+
+  describe('implicit button element', () => {
+    it('renders a link if you pass an href', () => {
+      const { $ } = render('button', {
+        href: '/'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.get(0).tagName).toEqual('a')
+      expect($component.attr('href')).toEqual('/')
```

Comment:
This doesn't relate to the test description. I think we should just be testing that it's rendering a link.